src/stores/CheckoutController.ts

```diff
@@ -134,19 +136,28 @@ export const directPostCardPayment = async (cardPaymentPayload: CardPaymentData)
   });
 };
 
-export const createAdyenPaymentSession = async (returnUrl: string): Promise<AdyenPaymentSession> => {
+export const createAdyenPaymentSession = async (returnUrl: string, isInitialPayment: boolean = true): Promise<AdyenPaymentSession> => {
   return await useAccount(async ({ auth: { jwt } }) => {
     return await useService(async ({ checkoutService, sandbox = true, authProviderId }) => {
       const { order } = useCheckoutStore.getState();
 
-      if (!order) throw new Error('No order created');
+      const getOrderIdIfInitialPayment = () => {
+        if (!isInitialPayment) return undefined;
+
+        if (!order) {
+          throw new Error('No order created');
+        }
+
+        return order.id;
+      };
+
```

Comment:
This can be simplified:

```suggestion
      const orderId = order?.id;
      
      if (isInitialPayment && !orderId) throw new Error('There is no order to pay for');

```

Reply:
What was I thinking......